Migrate to Nx monorepo

kai-nguyen-aligent opened the pull request:
This PR migrate `microservice-lib-util` to Nx monorepo setup with the following notable points:

- Setup Nx following our standards. This setup heavily inspired by our Nx Serverless template.
- Incoperate with our latest `ts-code-standard`.
- Added `verdaccio` to support local development.

Documents needs lots of update but I would like to open this PR for feedback gathering so we can draft out the document, especially the contribution & development guideline. Please ask question and suggestion for improvement.

jacob-ling-aligent reviewed (commented):
- I don't really understand how the package gets published.
  - Is this automated by a github action? If so, where is the config file for that? 
  -  Something like this looks useful: https://github.com/marketplace/actions/publish-to-npm
- verdaccio was mentioned to be removed from this PR and we can add it to another PR.
- Also a contributing doc was mentioned, would that go in the base directory and then link to it in each package?
- Does this repo need a license? Is there a standard for this here?

tvhees commented:
> * I don't really understand how the package gets published.
>   
>   * Is this automated by a github action? If so, where is the config file for that?
>   * Something like this looks useful: https://github.com/marketplace/actions/publish-to-npm
This does seem to be missing a publish action. I assume we can use something similar to https://github.com/aligent/serverless-conventions (eventually I'd like that tool to move to this repo as well).

> * verdaccio was mentioned to be removed from this PR and we can add it to another PR.
Correct. I'd like to be careful with 'nice to have' local dev tools in our shared repos.

> * Also a contributing doc was mentioned, would that go in the base directory and then link to it in each package?
> * Does this repo need a license? Is there a standard for this here?
There's shared code of conduct, contribution guide etc. here (https://github.com/aligent/code-of-conduct) but now that I think about it the license isn't shared.

Again, probably fine to use the same one as `serverless-conventions`.

@TheOrangePuff interested in DevOps guidance here.

tvhees reviewed (changes requested):
Looks good Kai. Agreed with Jacob's comments -

- Set up the github action, or add it as a followup task on the board
- Remove verdaccio
- Link to our Code of Conduct (https://github.com/aligent/code-of-conduct) and add a license where required

kai-nguyen-aligent commented:
@tvhees @jacob-ling-aligent I have addressed all of the feedback above. Please help me to double check.
- I'm not sure if the way I structure code of conducts and license is good enough or not.
- I also added the release Github Action, which is an adaptation of @TheOrangePuff 's `serverless-conventions` pipeline and Nx guideline: https://nx.dev/recipes/nx-release/publish-in-ci-cd
- I still believe that the level of documentation needs to be improved. However, let's address that in a future PR.

jacob-ling-aligent commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,57 @@
+
+## Issues
+
+If you identify and issue, please first search of the issue already exists, and provide further clarifications on that where possible. If the related issue doesn't exist, then please open one and provide as much information as possible.
```
If you identify an issue, please check if the issue already exists and provide further clarifications on the related issue where possible. If the related issue doesn't exist, then please open one and provide as much information as possible.

tvhees replied:
@jacob-ling-aligent should be a PR on the [central code of conduct repo](https://github.com/aligent/code-of-conduct/blob/main/CONTRIBUTING.md)

jacob-ling-aligent replied:
Oh right, because we want to link to it directly instead of copy pasting :+1:

jacob-ling-aligent replied:
I couldn't add a reviewer to the repo, but linking here for reference: https://github.com/aligent/code-of-conduct/pull/2

tvhees replied:
Merged in there :)

jacob-ling-aligent reviewed (changes requested):
LGTM, just a couple of sentences need rework in the contributing doc

tvhees commented on package.json:
```diff
@@ -1,48 +1,62 @@
+    "typedoc": "^0.26.11",
+    "typedoc-plugin-markdown": "^4.2.10",
+    "typescript": "^5.6.3",
+    "verdaccio": "^5.32.2",
```
@kai-nguyen-aligent double check npm dependencies

tvhees commented on CODE_OF_CONDUCT.md:
@kai-nguyen-aligent my current preference is we link to the central one for code of conduct and contributing, like we've done in the [mesh repository](https://github.com/aligent/aligent-graphql-mesh/blob/main/docs/CODE_OF_CONDUCT.md).

It just means we have one place to maintain the text instead of every repo.

tvhees reviewed (changes requested):
@kai-nguyen-aligent sorry if I wasn't clear enough about code of conduct! Some changes

kai-nguyen-aligent commented:
@tvhees @jacob-ling-aligent All of the feedback has been addressed:
- I removed `verdaccio` from `devDependencies`.
- Update Code of Conduct & Contribution following what we do in Mesh repo.
- Clean up `README.md`.

tvhees reviewed (approved):
@kai-nguyen-aligent good job, we're ok to merge this. Devops can review the pipeline afterwards.
